scripts/multi_ldap_change.php

```diff
@@ -0,0 +1,135 @@
+<?php
+#==============================================================================
+# Includes
+#==============================================================================
+require_once(__DIR__."/../conf/config.inc.php");
+require_once(__DIR__."/../lib/functions.inc.php");
+
+#==============================================================================
+# Action
+#==============================================================================
+
+$result = "";
+$return = Array();
+$error_code = 1;
+$login = $argv[1];
+$newpassword = $argv[2];
+$oldpassword = '';
+
+$log_file = fopen(__DIR__.'/multi_ldap_change.log', 'a+');
+fwrite($log_file, "Change '$login' password...\n");
+
+foreach ($secondaries_ldap as $s_ldap) {
+    # Connect to LDAP
+    $ldap = ldap_connect($s_ldap['ldap_url']);
+    ldap_set_option($ldap, LDAP_OPT_PROTOCOL_VERSION, 3);
+    ldap_set_option($ldap, LDAP_OPT_REFERRALS, 0);
+    if ( $ldap_starttls && !ldap_start_tls($ldap) ) {
+        $result = "ldaperror";
+        fwrite($log_file, "LDAP - Unable to use StartTLS");
+    } else {
+
+    # Bind
+    if ( isset($ldap_binddn) && isset($ldap_bindpw) ) {
+        $bind = ldap_bind($ldap, $ldap_binddn, $ldap_bindpw);
+    } else {
+        $bind = ldap_bind($ldap);
+    }
+
+    if ( !$bind ) {
+        $result = "ldaperror";
+        $errno = ldap_errno($ldap);
+        if ( $errno ) {
+        fwrite($log_file, "LDAP - Bind error $errno  (".ldap_error($ldap).")");
+        }
+    } else {
+
+    # Search for user
+    $ldap_filter = str_replace("{login}", $login, $ldap_filter);
+    $search = ldap_search($ldap, $ldap_base, $ldap_filter);
+
+    $errno = ldap_errno($ldap);
+    if ( $errno ) {
+        $result = "ldaperror";
+        fwrite($log_file, "LDAP - Search error $errno  (".ldap_error($ldap).")");
+    } else {
+
+    # Get user DN
+    $entry = ldap_first_entry($ldap, $search);
+    $userdn = ldap_get_dn($ldap, $entry);
+
+    if( !$userdn ) {
+        $result = "badcredentials";
+        fwrite($log_file, "LDAP - User $login not found");
+    } else {
+
+    # Get user email for notification
+    if ( $notify_on_change ) {
+        $mailValues = ldap_get_values($ldap, $entry, $mail_attribute);
+        if ( $mailValues["count"] > 0 ) {
+            $mail = $mailValues[0];
+        }
+    }
+
+    # Check objectClass to allow samba and shadow updates
+    $ocValues = ldap_get_values($ldap, $entry, 'objectClass');
+    if ( !in_array( 'sambaSamAccount', $ocValues ) and !in_array( 'sambaSAMAccount', $ocValues ) ) {
+        $samba_mode = false;
+    }
+    if ( !in_array( 'shadowAccount', $ocValues ) ) {
+        $shadow_options['update_shadowLastChange'] = false;
+        $shadow_options['update_shadowExpire'] = false;
+    }
+
```

Comment:
Maybe too complex to add this in this script. Because we don't know if secondary LDAP are Samba/AD/...

I think this script should just connect to the secondary LDAP, search user and change password.